### PR TITLE
Firebase email and password authentication in Android using Kotlin

### LINKED ISSUE #760: Firebase password and email authentication in Android using Kotlin
### Brief Summary: 
Firebase authentication is one of the most secure ways of controlling access to your app. Authenticated users can easily sync data between devices and analyze their app usage. This article will walk you through every step involved in setting up an Authentication system.

### Key Takeaways: 
By the end of this tutorial the reader should be able to:

1. Create a firebase project
2. Implement Firebase APIs
3. Control users accounts from firebase console

### References: 
N/A

### PR DESCRIPTION
**Pull request by Ericgacoki**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

 Fixes  #760

### CODE REVIEW

**Comment by hectorkambow** [comment]
👋 Thank you for submitting this PR - we will review it ASAP - all PRs go into our queue will be looked over before being approved to be published.

**Comment by kayere** [comment]
Hello @Ericgacoki, thank you for submitting your first article. You have covered a great topic. I will be reviewing your article. The review process consists of two stages: grammar and technical review. I will inform you when I find any errors. Once again, welcome to Section's engineering education program.

File: articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md [hunk 1]
```diff
@@ -0,0 +1,728 @@
...
+
+Before we proceed make sure you have the following plugin in your build. Gradle (app module level)
+
+```Kotlin
+plugins{
+       id 'kotlin-android-extensions'
+}
+```
+
+Inside CreateAccountActivity.kt, import kotlin synthetics like this
```
Comment (kayere) [inline articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md:506]: Add backticks to the file name.

File: articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md [hunk 2]
```diff
@@ -0,0 +1,728 @@
...
+
+- `extensions` : holds objects that entails Kotlin extension functions. This makes it possible to access and use such functions from any class in the app without creating another instance of their parent.
+
+- `utils`: contains utilities used in the app. In our case, this will only contain firebase utilities.
+
+- `views`: This contains the UI-related classes.  
+
+In advanced architectures like **MVVM**, there are more commonly used packages like the `model` but we're not concerned about them for now.
+
+Right click on views and add two empty activities namely `CreateAccountActivity` and `SignInActivity`. These two comes with their respective Xml layout files namely `activity_create_account` and `activity_sign_in`.
```
Comment (kayere) [inline articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md:167]: replace "Right click" with "Right-click".

File: articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+
+- [Android Studio IDE](https://developer.android.com/studio) installed in your machine
+- Knowledge on [constraint layout](https://developer.android.com/training/constraint-layout) in XML
+- An active [Google account](https://accounts.google.com/signup/v2/webcreateaccount?flowName=GlifWebSignIn&flowEntry=SignUp)
+
+### Step 1 -- Creating An Android project
+
+First, we need to create an Android project that we will use in the entire tutorial. To do this, launch the android studio and you'll see a welcome screen similar to the one below.
+
+![image](android_studio_welcome_screen.png "Android studio home screen")
```
Comment (kayere) [inline articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md:17]: Use this format for adding image links (/engineering-education/<article folder name>/<image name>)

**Review by kayere — COMMENTED** [review]
@Ericgacoki - Please make the following changes before we move to the next review stage.

**Comment by Ericgacoki** [comment]
Hello @kayere, thank you for the suggestions you've made so far. I'll fix them ASAP.

**Comment by Ericgacoki** [comment]
Hey @kayere, I've done the changes that you recommended.

File: articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md (hunk 3 above)
Comment (Ericgacoki) [reply articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md]: After doing this, images appear broken in the preview. Is it still okay ?

Comment (kayere) [reply articles/firebase-email-and-password-authentication-in-android-using-kotlin/index.md]: Yes, it is okay.

**Review by kayere — COMMENTED** [review]
Hello @Ericgacoki, I am done with tech review and I have one suggestion. Use hiphens instead of underscores in your image names.

**Comment by Ericgacoki** [comment]
Greetings @kayere, I have made the changes that you suggested. Please review.

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by hectorkambow** [comment]
Great first article @Ericgacoki  🚀  
Very nicely put together. Please be sure to go over the edits made by myself and our peer reviewer to further improve and apply any of these changes to future articles. 

There were a few minor issues with image paths. (see corrections).

**Pull request merged**